WordPress/Classes/Utility/ContextManager.swift

```diff
@@ -235,3 +245,9 @@ extension ContextManager {
         return sharedInstance()
     }
 }
+
+private enum SaveContextOption {
+    case synchronously
+    case asynchronously
+    case asynchronouslyWithCallback(completion: () -> Void, queue: DispatchQueue)
+}
```

Comment:
I like this and, as mentioned above, I could see us relying more on this construct to add some build-time railguards to how we run code on queues.